server/accounts.go

```diff
@@ -1344,26 +1345,85 @@ func (a *Account) AddServiceImportWithClaim(destination *Account, from, to strin
 		return ErrServiceImportAuthorization
 	}
 
-	if a.importFormsCycle(destination, from, to) {
-		return ErrServiceImportFormsCycle
+	// Check if this introduces a cycle before proceeding.
+	if a.serviceImportFormsCycle(destination, from) {
+		return ErrImportFormsCycle
```

Comment:
Will this cause a warning trace to help detect these issues?

Reply:
Yes I believe so..